pandas/io/sql.py

```diff
@@ -564,6 +564,10 @@ def to_sql(frame, name, con, flavor='sqlite', schema=None, if_exists='fail',
         raise NotImplementedError("'frame' argument should be either a "
                                   "Series or a DataFrame")
 
+    if dtype and not isinstance(dtype,dict):
+        temp_type = dtype
+        dtype = { col_name : temp_type for col_name in frame }
```

Comment:
Also `is_dictlike` (which isn't a perfect function but at least improvements will be encapsulated)